qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1988 @@
+
+        with pulse.build() as pulse_prog:
+            pulse.play(pulse.Constant(100,0.5), d0)
+
+        # execute on real backend
+        # qiskit.execute(pulse_prog, backend)
+
+    Args:
+        backend (BaseBackend): A Qiskit backend. If not supplied certain
+            builder functionality will be unavailable.
+        schedule: a *mutable* pulse Schedule in which your pulse program will
+            be built.
+        default_alignment: Default scheduling alignment for builder.
+            One of ``left``, ``right``, ``sequential`` or an alignment
+            contextmanager.
+        default_transpiler_settings: Default settings for the transpiler.
+        default_circuit_scheduler_settings: Default settings for the
+            circuit to pulse scheduler.
+
+    Returns:
+        A new builder context which has the active builder inititalized.
+    """
+    return _PulseBuilder(
+        backend=backend,
+        schedule=schedule,
+        default_alignment=default_alignment,
+        default_transpiler_settings=default_transpiler_settings,
+        default_circuit_scheduler_settings=default_circuit_scheduler_settings)
+
+
+# Builder Utilities ############################################################
+class NoActiveBuilder(exceptions.PulseError):
+    """Raised if no builder context is active."""
+
+
+def _active_builder() -> _PulseBuilder:
+    """Get the active builder in the active context.
+
+    Returns:
+        The active active builder in this context.
+
+    Raises:
+        NoActiveBuilder: If a pulse builder function is called outside of a
+            builder context.
+    """
+    try:
+        return BUILDER_CONTEXTVAR.get()
+    except LookupError:
+        raise NoActiveBuilder(
+            'A Pulse builder function was called outside of '
+            'a builder context. Try calling within a builder '
+            'context, eg., "with pulse.build() as schedule: ...".')
+
+
+def active_backend():
+    """Get the backend of the currently active builder context.
+
+    Returns:
+        BaseBackend: The active backend in the currently active
+            builder context.
+
+    Raises:
+        exceptions.BackendNotSet: If the builder does not have a backend set.
+    """
+    builder = _active_builder().backend
+    if builder is None:
+        raise exceptions.BackendNotSet(
+            'This function requires the active builder to '
+            'have a "backend" set.')
+    return builder
+
+
+def append_block(block: Schedule):
+    """Call a block by appending to the active builder block."""
+    _active_builder().append_block(block)
+
+
+def append_instruction(instruction: instructions.Instruction):
+    """Append an instruction to the current active builder context block.
+
+    Example Usage:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+
+        d0 = pulse.DriveChannel(0)
+
+        with pulse.build() as pulse_prog:
+            pulse.append_instruction(pulse.Delay(10, d0))
```

Comment:
What's the use case for exposing this method?

Reply:
If a user defines their own instruction. This is the path to extending the builder currently.